OneSignalSDK/onesignal/core/src/main/java/com/onesignal/internal/OneSignalImp.kt

```diff
@@ -234,21 +246,10 @@ internal class OneSignalImp : IOneSignal, IServiceProvider {
                 configModel!!.disableGMSMissingPrompt = _disableGMSMissingPrompt!!
             }
 
-            // "Inject" the services required by this main class
-            _location = services.getService()
-            _user = services.getService()
-            _session = services.getService()
-            iam = services.getService()
-            _notifications = services.getService()
-            operationRepo = services.getService()
-            propertiesModelStore = services.getService()
-            identityModelStore = services.getService()
-            subscriptionModelStore = services.getService()
-            preferencesService = services.getService()
-
-            // Instantiate and call the IStartableServices
-            startupService = services.getService()
-            startupService!!.bootstrap()
+            // bootstrap services
+            for (bootstrapService in services.getAllServices<IBootstrapService>()) {
+                bootstrapService.bootstrap()
+            }
```

Comment:
Why was `startupService!!.bootstrap()` removed and replaced with this for loop?
* `startupService` abstracted this away so I think it was clearer before.

Reply:
Calling startupService within initWithContext causes the serviceProvider to resolve all bootstrap and startable services right away. In my benchmark calling getService for startupService contributes around 80% of execution time in initWithContext. So the idea of this PR is to prevent startupService from resolving all services immediately and instead resolve the unused services in a background thread. 

Will making startupService a different object other than service makes more sense? So we can manage bootstrap and start there without resolving the services.

Reply:
ah right, my mistake. We can probably leave it here then, since it is only 3 lines.